manage

```diff
@@ -275,13 +293,26 @@ elif [ "$1" == 'update' ];then
   cp -rf "${DIRECTORY}/update/pi-apps/apps/${app}" "${DIRECTORY}/apps/${app}"
   
   if [ "$installback" == 'yes' ];then
-    echo "$app was originally installed before updating. Reinstalling the new version now."
     #install it using a recursive script instance
     "${DIRECTORY}/manage" install "$app"
+    if [ $? == 0 ]; then
+      failed=true
+    else
+      failed=false
+    fi
+  fi
+  
+  if [ "$failed" == 'true' ]; then
+    error "Failed to update ${app}."
```

Comment:
This breaks the script because the `error ` function runs `exit 1`. In this case, when the first app fails to update, subsequent appd wouldn't be reinstalled at all, because the usage of `error()` would kill the script prematurely.